src/app/archive/create_schema_new.sql

```diff
@@ -0,0 +1,73 @@
+CREATE TABLE state_hashes
+( id    serial PRIMARY KEY
+, value text   NOT NULL UNIQUE
+);
+
+CREATE INDEX idx_state_hashes_value ON state_hashes(value);
+
+CREATE TABLE public_keys
+( id    serial PRIMARY KEY
+, value text   NOT NULL UNIQUE
+);
+
+CREATE INDEX idx_public_keys_value ON public_keys(value);
+
+CREATE TABLE snarked_ledger_hashes
+( id    serial PRIMARY KEY
+, value text   NOT NULL UNIQUE
+);
+
+CREATE INDEX idx_snarked_ledger_hashes_value ON snarked_ledger_hashes(value);
+
+CREATE TYPE user_command_type AS ENUM ('payment', 'delegation');
+
+CREATE TABLE user_commands
```

Comment:
Both `user_commands` and `internal_commands` require a field `transaction_id int NOT NULL REFERENCES transactions(id)` in order to conform with the interface @figitaki requested.

Reply:
I don't believe so, the use case for the `transactions` table was to enable querying for all the `user_commands` and `internal_commands` from a particular block which the given schema achieves. @Schmavery also suggested simply adding a block reference to the `user_commands` table and the `internal_commands` which might solve this problem as well and remove the need for a `transactions` table entirely.

Reply:
Also both of these tables seem to be missing `hash` fields, but not sure if that's part of the spec.

Reply:
A `user_command` might be included in multiple different blocks, so it's not a good choice to have `user_command` refers to `block` directly. We need have a join table for `block` and `user_command`.
The same argument also applies for `internal_command`.